lib/health_checks/sidekiq.rb

```diff
@@ -36,16 +40,29 @@ def start
       end
     end
 
-    def check_redis
-      sidekiq_response = ::Sidekiq.redis(&:ping)
-      return if sidekiq_response == 'PONG'
+    def create_clients(mongo_databases)
+      mongo_databases.map do |db|
+        client_name = "#{db[:name]}_sidekiq_health_check"
+        client = MongoidCustomClient.create(client_name, db[:hosts])
+
+        { db_name: db[:name], instance: client }
+      end
+    end
 
-      raise "Sidekiq.redis.ping returned #{sidekiq_response.inspect} instead of PONG"
+    def check_mongoid_clients(clients)
+      db_name = ''
+      clients.each do |client|
+        db_name = client[:db_name]
+        client[:instance].command(dbStats: 1).first['db']
+      end
+    rescue => e
+      raise "#{e}, Database Name: #{db_name}"
     end
 
-    def check_mongo
-      session = Mongoid::Clients.with_name(:default)
-      session.database_names
+    def check_redis_connections(redis_configs)
+      redis_configs.each do |config|
```

Comment:
Why not the same wrap as above here? How would you know which Redis instance had problems?

Reply:
yes, I will. Just to figure out how to pass the db name in a better way